pennylane/devices/experimental/default_qubit_2.py

```diff
@@ -197,7 +221,19 @@ def execute(
             self.tracker.update(batches=1, executions=len(circuits))
             self.tracker.record()
 
-        results = tuple(simulate(c, rng=self._rng, debugger=self._debugger) for c in circuits)
+        max_workers = self._get_max_workers(execution_config)
+        if max_workers is None:
+            results = tuple(simulate(c, rng=self._rng, debugger=self._debugger) for c in circuits)
+        else:
+            vanilla_circuits = [convert_to_numpy_parameters(c) for c in circuits]
+            _wrap_simulate = partial(simulate, rng=self._rng, debugger=self._debugger)
```

Comment:
Also, I think we should not support the snapshots with multiprocessing.  I'm fine with just making sure that we don't have snapshots in the circuit if there's more than one worker.

Snapshots work with in-place modification of the `_debugger` object. Pretty sure that won't work with multi-processing.

Reply:
I added a function `_validate_multiprocessing_circuits` following your suggestion, and a test. Let me know what you think.